ras_party/controllers/queries.py

```diff
@@ -643,3 +643,23 @@ def count_enrolment_by_survey_business(business_id, survey_id, session):
         .count()
     )
     return response
+
+
+def query_enrolments_by_parameters(
+    session: session, respondent_id: int = None, business_id: UUID = None, survey_id: UUID = None, status: int = None
+) -> list[Enrolment]:
+    """
+    Query to return a list of enrolments based on parameters
+    """
+    conditions = []
+
+    if respondent_id:
+        conditions.append(Enrolment.respondent_id == respondent_id)
+    if business_id:
+        conditions.append(Enrolment.business_id == business_id)
+    if survey_id:
+        conditions.append(Enrolment.survey_id == survey_id)
+    if status:
+        conditions.append(Enrolment.status == status)
+
+    return session.query(Enrolment).filter(and_(*conditions)).all()
```

Comment:
Could this return to much and need restricting with a count, maybe. It depends on how it is used. Nearly ever use I see are respondent/business driven which means this should be fine. Calling to it with just survey could produce a large result set, but I don't believe we every do that. As said above a place I might harden after I start plugging it in

Reply:
As we can see from other API, there is a danger that other developers misuse or don't recognise the implications of calling this in future solutions. Again, at the very least should caveat its usage in its definition.

Reply:
We do need to trust developers and the review process even if it has broken in the past, otherwise we are struggling. As I said above I do intend on hardening this, but I can't stop future missus completely